### PR TITLE
Add test coverage for modal animation

### PR DESCRIPTION
**Pull request by jelhan**
This adds test coverage for modal animation. It encodes the results of my investigation on the topic documented in https://github.com/kaliber5/ember-bootstrap/pull/1402/files#r605945018. I thought it might be helpful to get this in before #1402. And maybe even if I don't find the time to finish that one.

### CODE REVIEW

**Comment by jelhan** [comment]
Tests are failing for `ember-beta` and `ember-canary`. Not related to the additional test added here.

File: tests/integration/components/bs-modal-test.js [hunk 1]
```diff
@@ -246,4 +246,51 @@ module('Integration | Component | bs-modal', function (hooks) {
...
+    await settled();
+    assert.dom('.modal').doesNotExist();
+  });
+
+  testBS4('it animates opening and closing the modal', async function (assert) {
+    this.set('open', false);
+    await render(hbs`<BsModal @open={{this.open}}>Hello world!</BsModal>`);
+
+    this.set('open', true);
+    await waitFor('.modal.show');
```
Comment (simonihmig) [inline tests/integration/components/bs-modal-test.js:278]: I should have commented here in the first place, but did in the other PR: https://github.com/kaliber5/ember-bootstrap/pull/1402/files#r623753413

**Comment by jelhan** [comment]
I added assertion in the tests that a transition is run for opening and closing by listening to `transitionrun` event as proposed here: https://github.com/kaliber5/ember-bootstrap/pull/1402#discussion_r634727211 It works very well for opening transition in Bootstrap 4. But no closing transition is logged. For Bootstrap 3 it breaks for both transitions. In both cases it seems as if not transition is run. At least the events are not emitted. Will debug further.

**Comment by jelhan** [comment]
I finally figure a way out to test this reliable in Bootstrap 4: `el.getAnimations()`. It works both for opening and closing and allows to write readable code.

I decided to not write tests for Bootstrap 3. In Bootstrap 3 I was neither able to log transition events nor to get any animation using `el.getAnimations()`. I did not wanted to invest even more time into writing this test. While I learnt _a lot_ about CSS transitions and related JavaScript APIs, it also took quiet a while. I think not having test coverage for Bootstrap 3 is acceptable as we plan to deprecate support for it soonish. Until then we can rely on manual tests of modal animations in Bootstrap 3.

File: tests/integration/components/bs-modal-test.js [hunk 2]
```diff
@@ -246,4 +246,65 @@ module('Integration | Component | bs-modal', function (hooks) {
...
     assert.dom('.modal').isVisible('Modal is visible again');
   });
+
+  testNotBS3('it animates opening the modal', async function (assert) {
+    this.set('open', false);
+    await render(hbs`<BsModal @open={{this.open}}>Hello world!</BsModal>`);
+
+    this.set('open', true);
+    await waitFor('.modal.show');
+    assert.dom('.modal').hasClass('show');
```
Comment (simonihmig) [inline tests/integration/components/bs-modal-test.js:256]: This assertion is a bit meaningless, given the previous line. But I don't really care :)

File: tests/integration/components/bs-modal-test.js [hunk 3]
```diff
@@ -246,4 +246,65 @@ module('Integration | Component | bs-modal', function (hooks) {
...
+
+    // wait until modal is shown and opening transition is finished
+    await waitFor('.modal.show');
+    await Promise.all(
+      find('.modal')
+        .getAnimations()
+        .map((animation) => animation.finished)
+    );
+
+    // keep reference to modal element to compare with event target after modal has been destroyed
```
Comment (simonihmig) [inline tests/integration/components/bs-modal-test.js:285]: Seems `modalEl` is not used "after modal has been destroyed", so this comment looks a bit misleading, or do I misunderstand?

**Review by simonihmig — APPROVED** [review]
Nice work, thanks so much!

TIL about `.getAnimations()` 😯

**Comment by simonihmig** [comment]
@jelhan so I already approved this. If you think my comments don't need to be addressed (they are super minor), then feel free to merge this right away!

**Pull request merged**